org.adempiere.base/src/org/compiere/acct/Doc.java

```diff
@@ -707,12 +721,36 @@ else if (isPosted())
 	 *	@return number of records deleted
 	 */
 	protected int deleteAcct()
-	{
-		StringBuilder sql = new StringBuilder ("DELETE FROM Fact_Acct WHERE AD_Table_ID=")
-			.append(get_Table_ID())
-			.append(" AND Record_ID=").append(p_po.get_ID())
-			.append(" AND C_AcctSchema_ID=").append(m_as.getC_AcctSchema_ID());
-		int no = DB.executeUpdate(sql.toString(), getTrxName());
+	{				
+		// backup the posting records before delete them
+		StringBuilder sql = new StringBuilder ("INSERT INTO T_Fact_Acct_History ")
+				.append("SELECT * FROM Fact_Acct ")
+				.append("WHERE AD_Table_ID=?")
+				.append(" AND Record_ID=?")
+				.append(" AND C_AcctSchema_ID=?");
+		int no = DB.executeUpdate(sql.toString(), new Object[] {get_Table_ID(), p_po.get_ID(), m_as.getC_AcctSchema_ID()}, false, getTrxName());
+		if (no != 0)
+			if (log.isLoggable(Level.INFO)) log.info("inserted=" + no);
+		
+		// set the updated to current time - for house keeping purpose
+		sql = new StringBuilder ("UPDATE T_Fact_Acct_History ")
+				.append("SET Updated=? ")
+				.append("WHERE Created=Updated ")
+				.append(" AND AD_Table_ID=?")
+				.append(" AND Record_ID=?")
+				.append(" AND C_AcctSchema_ID=?");

```

Comment:
make sure you have appropriate index create for this, otherwise it would be slow

Reply:
Created a database index: t_fact_acct_history_update